map/bookmark_manager.cpp

```diff
@@ -1141,6 +1142,53 @@ kml::CompilationType BookmarkManager::GetCompilationType(kml::MarkGroupId id) co
   return compilation->second->GetCategoryData().m_type;
 }
 
+void BookmarkManager::SaveTrackRecording(std::string const & name)
+{
+  CHECK_THREAD_CHECKER(m_threadChecker, ());
+  auto const & tracker = GpsTracker::Instance();
+  CHECK(tracker.IsRecentTrackCollectionInitialized(), ("Track should be initialized"));
+  std::vector<location::GpsTrackInfo> recentTrack = tracker.GetRecentTrackCollection();
+  CHECK(!recentTrack.empty(), ("Recorded track should not be empty"));
+
+  kml::TrackData trackData;
+  std::string trackName = name.empty() ? GenerateTrackRecordingName() : name;
+  kml::SetDefaultStr(trackData.m_name, trackName);
+
+  struct kml::MultiGeometry m_geometry;
+  m_geometry.FromGpsInfoPoints(recentTrack);
+  trackData.m_geometry = std::move(m_geometry);
+
+  struct kml::ColorData colorData;
+  colorData.m_rgba = GenerateTrackRecordingColor().GetRGBA();
+  struct kml::TrackLayer layer;
+  layer.m_color = colorData;
+
+  std::vector<kml::TrackLayer> m_layers;
+  m_layers.emplace_back(layer);
+  trackData.m_layers = std::move(m_layers);
+  trackData.m_timestamp = kml::TimestampClock::now();
+
+  auto editSession = GetEditSession();
+  auto const track = editSession.CreateTrack(std::move(trackData));
+  auto const groupId = LastEditedBMCategory();
+  AttachTrack(track->GetId(), groupId);
+}
+
+std::string BookmarkManager::GenerateTrackRecordingName() const
+{
+  /// @todo: - improve the track name generation
```

Comment:
Use capital letters (they're used in some IDEs), and leave your initials to understand who left it without reading the git log.
```suggestion
  /// @TODO(KK): improve the track name generation
```

Reply:
non-capital letters todo are used exclusively by @vng , so everyone else should use TODO(XY) instead, according to Google C++ spec.